crates/config/src/config.rs

```diff
@@ -224,7 +229,8 @@ impl Default for SenderRecoveryConfig {
 }
 
 /// Execution stage configuration.
-#[derive(Debug, Clone, Copy, Deserialize, PartialEq, Eq, Serialize)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 #[serde(default)]
```

Comment:
we also need to cfg_attr this like

#[cfg_attr(feature = "serde", serde(default))]